x/dex/exchange/settlement.go

```diff
@@ -16,6 +16,7 @@ func Settle(
 	order types.OrderBook,
 	takerDirection types.PositionDirection,
 	worstPrice sdk.Dec,
+	isTakerLiquidation bool,
```

Comment:
can we just pass in the takerOrderType (e.g. `types.OrderType_LIQUIDATION`) here? so that we don't need to do the check below

Reply:
also can we add a general comment of the settlement logic in this function

Reply:
added